src/Http/Client/Adapter/Curl.php

```diff
@@ -110,9 +110,10 @@ public function buildOptions(RequestInterface $request, array $options): array
         }
 
         $body = $request->getBody();
-        $body->rewind();
-        $out[CURLOPT_POSTFIELDS] = $body->getContents();
```

Comment:
Could you unset the field key if the value has no length?

Reply:
Pushed up a commit that just unsets `CURLOPT_POSTFIELDS` based on this comment.